Error with pre-restore-database during upgrade

Project:

* GeoMapFish version: **2.1.2.dev1**

Running command: ` make -f instance_v2.mk upgrade4`

Result:

```
.build/venv/bin/c2ctool upgrade  instance_v2.mk --step 4
make[1]: Entering directory 'v2'
.build/venv/bin/c2c-template --vars vars_nyon.yaml --engine mako --files CONST_create_template/deploy/hooks/pre-restore-database.mako
Traceback (most recent call last):
  File ".build/venv/bin/c2c-template", line 11, in <module>
    sys.exit(main())
  File "v2/.build/venv/local/lib/python2.7/site-packages/c2c/template.py", line 213, in main
    _proceed(files, used_vars, options)
  File "v2/.build/venv/local/lib/python2.7/site-packages/c2c/template.py", line 237, in _proceed
    bottle_template(files, used_vars, engine)
  File "v2/.build/venv/local/lib/python2.7/site-packages/c2c/template.py", line 268, in bottle_template
    template, **used_vars
  File "v2/.build/venv/bin/bottle.py", line 3619, in template
    return TEMPLATES[tplid].render(kwargs)
  File "v2/.build/venv/bin/bottle.py", line 3290, in render
    return self.tpl.render(**_defaults)
  File "v2/.build/venv/local/lib/python2.7/site-packages/mako/template.py", line 462, in render
    return runtime._render(self, self.callable_, args, data)
  File "v2/.build/venv/local/lib/python2.7/site-packages/mako/runtime.py", line 838, in _render
    **_kwargs_for_callable(callable_, data))
  File "v2/.build/venv/local/lib/python2.7/site-packages/mako/runtime.py", line 873, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "v2/.build/venv/local/lib/python2.7/site-packages/mako/runtime.py", line 899, in _exec_template
    callable_(context, *args, **kwargs)
  File "CONST_create_template_deploy_hooks_pre_restore_database_mako", line 25, in render_body
TypeError: 'Undefined' object has no attribute '__getitem__'
CONST_Makefile:658: recipe for target 'CONST_create_template/deploy/hooks/pre-restore-database' failed
make[1]: *** [CONST_create_template/deploy/hooks/pre-restore-database] Error 1
make[1]: Leaving directory 'v2'
Traceback (most recent call last):
  File ".build/venv/bin/c2ctool", line 11, in <module>
    sys.exit(main())
  File "v2/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 100, in main
    c2ctool.upgrade()
  File "v2/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 231, in upgrade
    self.step4()
  File "v2/.build/venv/local/lib/python2.7/site-packages/c2cgeoportal/scripts/c2ctool.py", line 452, in step4
    check_call(["make", "-f", self.options.file, "build"])
  File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['make', '-f', 'instance_v2.mk', 'build']' returned non-zero exit status 2
CONST_Makefile:1284: recipe for target 'upgrade4' failed
make: *** [upgrade4] Error 1
```

If no direct error, result and expected result:

--- a/c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl
+++ b/c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl
@@ -8,6 +8,14 @@ Version 2.1.2
 
     git mv deploy/hooks/post-{create,restore}
 
+2. Add in the project vars file:
+
+    deploy:
+        code_source: /var/www/c2cgeoportal_mapfish/private/c2cgeoportal
+        code_destination: /var/www/c2cgeoportal_mapfish/private/c2cgeoportal
+
+    with the right deploy path.
+
 
 Version 2.1.0
 =============

--- a/c2cgeoportal/scripts/c2ctool.py
+++ b/c2cgeoportal/scripts/c2ctool.py
@@ -351,6 +351,9 @@ class C2cTool:
             "{0!s}/pcreate".format(self.venv_bin), "--ignore-conflicting-name", "--overwrite",
             "--scaffold=c2cgeoportal_update", "../{0!s}".format(self.project["project_folder"])
         ])
+        for deploy_hook_file in os.listdir("CONST_create_template/deploy/hooks"):
+            check_call(["chmod", "+x", "CONST_create_template/deploy/hooks/{}".format(deploy_hook_file)])
+
         check_call(["make", "-f", self.options.file, self.options.clean])
 
         # Update the package.json file

--- a/c2cgeoportal/scripts/c2ctool.py
+++ b/c2cgeoportal/scripts/c2ctool.py
@@ -346,6 +346,7 @@ class C2cTool:
                 pip_cmd += ["c2cgeoportal=={0!s}".format((self.options.version))]
             check_call(pip_cmd)
 
+        shutil.rmtree("CONST_create_template")
         check_call([
             "{0!s}/pcreate".format(self.venv_bin), "--ignore-conflicting-name", "--overwrite",
             "--scaffold=c2cgeoportal_update", "../{0!s}".format(self.project["project_folder"])

--- a/travis/create-new-project.sh
+++ b/travis/create-new-project.sh
@@ -7,6 +7,9 @@ cp travis/build.mk /tmp/travis/testgeomapfish/travis.mk
 cp travis/vars.yaml /tmp/travis/testgeomapfish/vars_travis.yaml
 cd /tmp/travis/testgeomapfish
 
+chmod +x CONST_create_template/deploy/hooks/*
+chmod +x deploy/hooks/*
+
 echo "REQUIREMENTS = --editable ${TRAVIS_FOLDER}" | cat - travis.mk > travis.mk.new
 mv travis.mk.new travis.mk
 

